tests/integration/utils.go

```diff
@@ -2331,6 +2338,21 @@ func skipIfNetworkTest(t testing.TB, actions []any) {
 	}
 }
 
+// skipVectorEmbeddingTest skips the current test if the given actions
+// contain a schema with vector embedding generation and skipVectoEmbeeddingTest is true.
+func skipIfVectorEmbeddingTest(t testing.TB, actions []any) {
+	hasVectorEmbedding := false
+	for _, act := range actions {
+		switch a := act.(type) {
+		case SchemaUpdate:
+			hasVectorEmbedding = strings.Contains(a.Schema, "@embedding")
+		}
+	}
+	if !runVectorEmbeddingTests && hasVectorEmbedding {
+		t.Skip("test involves vector embedding generation")
+	}
+}
```

Comment:
todo: I dislike the automatic skipping based on detecting if it is a vector embedding test because this means we can not test the behavior that should happen when vector embedding is being used by the user but it was turned off (which as brought up by Kyle in the standup was an important behavior for us to ensure).

Instead I would like a usage like the `SupportedType` within the individual tests that specify that the tests are only testing the behavior for when "vector embedding is enabled" (Note: all your tests currently are only testing this behavior)

todo: Once above is done, I would add counter part tests to all (or most) embedding tests that were introduced in this PR and assert their behavior for when "vector embedding is turned off".

Reply:
same as above, as discussed in the standup, we will look ad changing the trigger for embedding tests in a subsequent PR.